File: seed/pydantic-v2/trace/resources/submission/types/trace_response_v_2.py
from pydantic import BaseModel
from uuid import UUID
from resources.submission.types.traced_file import TracedFile
from typing import Optional
from resources.commons.types.debug_variable_value import DebugVariableValue
from resources.submission.types.expression_location import ExpressionLocation
from resources.submission.types.stack_information import StackInformation
from dt import datetime
from core.datetime_utils import serialize_datetime
class TraceResponseV2(BaseModel):
    submission_id: UUID = 
    line_number: int = 
    file: TracedFile
    return_value: Optional[DebugVariableValue] = 
    expression_location: Optional[ExpressionLocation] = 
    stack: StackInformation
    stdout: Optional[str] = None
    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}

File: seed/ruby-sdk-v2/trace/lib/seed/submission/trace_response_v_2.rb

module Seed
    module Types
        class TraceResponseV2 < Internal::Types::Model
            field :submission_id, String, optional: false, nullable: false
            field :line_number, Integer, optional: false, nullable: false
            field :file, Seed::submission::TracedFile, optional: false, nullable: false
            field :return_value, Seed::commons::DebugVariableValue, optional: true, nullable: false
            field :expression_location, Seed::submission::ExpressionLocation, optional: true, nullable: false
            field :stack, Seed::submission::StackInformation, optional: false, nullable: false
            field :stdout, String, optional: true, nullable: false
        end
    end
end
